Add gallery view for search results

jxjj opened the pull request:
This adds a gallery view of search results:
- The gallery creates slides for search matches, any files within the asset (that is, any objects with a defined `fileId`), and any related assets items found in the asset's `relatedAssetCache`. 
- While adjacent results are loading asynchronously, placeholder slides will be used.  
- When the `activeSlideIndex` is within 10 of the last slide, the gallery will load more search results.

(At least) one bug that needs to be resolved: with some searches on dev, like "Test", the gallery with throw an error and the active slide index won't update properly in the main slider. I'm probably making an incorrect assumption about the shape or existence of some match result or asset data.

Currently up on dev for testing: 
https://dev.elevator.umn.edu/defaultinstance/search/s/7eeb0e19-1f48-4806-a22c-238c2aa39199?resultsView=gallery

<img width="500" alt="image" src="https://user-images.githubusercontent.com/980170/235742341-0080e1b9-8310-4bf5-b66f-d1f5471d9298.png">

Closes #58

cmcfadden commented on src/router.ts:
```diff
@@ -31,6 +31,10 @@ const routesForTesting: RouteRecordRaw[] = [
     }),
   },
+  {
+    path: "/test/slides",
```
do we want to keep this route?

jxjj replied:
dropped

cmcfadden commented:
Is a fullscreen mode still on your todo for this, or am I missing it?

One issue I'm seeing is that if you grab the slider at the bottom and move, it indicates it's changing the active item, but doesn't actually? Is that an easy-ish fix?

![May-02-2023 13-34-21](https://user-images.githubusercontent.com/211655/235755048-7da06646-548f-4662-8979-200876901a15.gif)

jxjj commented:
Thanks for the feedback and testing. I think this is ready for another round of review:

Changes:
- Fixed issue where gallery would sometimes throw. Turns out I made one too many placeholder slides.
- Added fullscreen toggle.
- Shifted layout a bit to accommodate fullscreen. It was getting busy at the top, and the prev/next there were redundant with the prev/next by the thumbs at the bottom, so I dropped the top buttons.
- Now sliding thumbs will update the main slide properly. One caveat: we're waiting for the drag to complete before updating the main slider to avoid loading extra object viewers.
- Added keyboard navigation with left/right arrows.

<img width="500" alt="image" src="https://user-images.githubusercontent.com/980170/235792994-95b26911-aedd-4625-b85a-c78f52af1dca.png">

jxjj commented:
Closes:
- #58

cmcfadden commented:
looks great!
